pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -479,6 +484,10 @@ stdenv.mkDerivation rec {
     echo "build --override_repository=${remote_java_tools.name}=${remote_java_tools}" > $out/etc/bazelrc
     echo "build --distdir=${distDir}" >> $out/etc/bazelrc
 
+    # we set a custom bazelrc in $out/etc/bazelrc
+    # the following "try-import" restores the read of the system-wide bazel.bazelrc
+    echo "try-import /etc/bazel.bazelrc" >> $out/etc/bazelrc
```

Comment:
Maybe a stupid question, but can we stop using `$out/etc/bazelrc` entirely, and instead add `--override_repository=${remote_java_tools.name}=${remote_java_tools}"` and `--distdir=${distDir}` to the bazel wrapper already present at https://github.com/NixOS/nixpkgs/pull/73309/files#diff-4746b81564a01e241a84d455575437e7R478?

Reply:
Well, plus that part:
```
      build --distdir=${distDir}
      fetch --distdir=${distDir}
      build --copt="$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --copt="/g')"
      build --host_copt="$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --host_copt="/g')"
      build --linkopt="-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --linkopt="-Wl,/g')"
      build --host_linkopt="-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --host_linkopt="-Wl,/g')"
      build --host_javabase='@local_jdk//:jdk'   
      build --host_java_toolchain='${javaToolchain}'
```

and dropping the
```
      # bazel reads its system bazelrc in /etc
      # override this path to a builtin one
      substituteInPlace \
        src/main/cpp/option_processor.cc \
        --replace BAZEL_SYSTEM_BAZELRC_PATH "\"$out/etc/bazelrc\""
```

Reply:
It'd probably also make sense to separate the `strict_action_env` fixes from the bazelrc improvements - so they should at least be in 2 separate commits.

Reply:
> Maybe a stupid question, but can we stop using `$out/etc/bazelrc` entirely, and instead add `--override_repository=${remote_java_tools.name}=${remote_java_tools}"` and `--distdir=${distDir}` to the bazel wrapper already present at https://github.com/NixOS/nixpkgs/pull/73309/files#diff-4746b81564a01e241a84d455575437e7R478?

Not possible. You can see that the options are in the `build` configuration section. However the wrapper does not know that we are using the `build` or another command. Moving this to the wrapper will need more logic, such as a detection of the command used by the bazel user.

The second part exists just at build time, that's the bazelrc of the buildtime bazel.

I do agree with the split. Done.

Reply:
Sorry for not being clear. I meant to split the whole bazelrc stuff in a followup PR.